framework/web/View.php

```diff
@@ -604,7 +604,7 @@ protected function renderBodyEndHtml($ajaxMode)
                 $lines[] = Html::script($js, ['type' => 'text/javascript']);
             }
             if (!empty($this->js[self::POS_LOAD])) {
-                $js = "jQuery(function ($) {\n$(window).on('load', function () {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});\n});";
+                $js = "jQuery(window).on('load', function () {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});";
```

Comment:
Hmm... Shouldn't it be just `"jQuery(function($) {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});"`?

Reply:
jQuery(function($) { }); - when DOM is ready
jQuery(window).on('load', function() { }); - when content is fully loaded

Reply:
Ah, OK.

Reply:
I see that we are nesting the calls unnecessarily here but as far as I understand DOM ready event comes before window load event. if we register the handler inside of the DOM ready event, why does it have no effect?

Reply:
In version 3 the ready handler fires asynchronously.

You can check code:

```javascript
jQuery(function ($) {
  console.log('ready');
  $(window).on('load', function () {
    console.log('load in ready');
  });
});
jQuery(window).on('load', function () {
  console.log('load');
});
```

With jQuery 2.2.4 you will see:

```
ready
load
load in ready
```

With jQuery 3.2.1 you will see:

```
load
ready
```

No "load in ready" anymore.

https://github.com/jquery/jquery/issues/3194#issuecomment-228556922
https://github.com/jquery/jquery/issues/3197

Reply:
I just wanted to make `$` working even jQuery.noConflict() is enabled. But missed this great change in jQuery API. The best way to fix this bug - rollback my proposal in #14620